src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -161,9 +161,6 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
         {
             var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails);
 
-            Debug.Assert(modeResult.FullMatches.Select(match => match.Mode).Distinct().Count() <= 1,
-                $"There should only be one mode match, check the {nameof(ModeDetails)}");
```

Comment:
that is, unless `asp-fallback-href="fall.css" asp-fallback-href-include="**/back.css" asp-fallback-test...` would (with all the test attributes provided) result in two full fallback matches.  if that's the case, would also need to prioritize matches within a mode.